### PR TITLE
Incremental PCA

### PR DESCRIPTION
**Pull request by jalabort**
This PR adds the method `increment`  to the `PCAModel` class.

It also:
- Add `ipca` function in `menpo.math.decomposition`
- Change name of function `principal_component_decomposition` to `pca`
- Remove `bias` and `whiten` kwargs from `pca` function.

The code is fairly stable and seems to be working as expected. The code can be easily tested using the following two notebooks on my `scrap` repo: `IPCA` and `IPCA - Menpo Model`.

However, before this is merged in, I want to clean up the `PCAModel` class a bit since there are things that are not correct at the moment. 

To do:
- [x] Add tests for `ipca`
- [x] Add tests for `increment` on `PCAModel`

### CODE REVIEW

**Comment by jalabort** [comment]
This is ready for inspection. 

I believe things like building a large `PCAModel` using the method `increment` from a potentially very large generator of high dimensional features using a `function` or a `class method` in `PCAModel` should be handled in a separate PR.

**Comment by patricksnape** [comment]
@jabooth Are we happy with this, shall we get it in?

**Comment by jabooth** [comment]
Haven't actually reviewed it yet - I'll look it over today

File: menpo/math/decomposition.py [hunk 1]
```diff
@@ -56,96 +60,178 @@ def principal_component_decomposition(X, whiten=False, centre=True,
...
-    whiten : `bool`, optional
-        Normalise the eigenvectors to have unit magnitude.
-    centre : `bool`, optional
-        Whether to centre the data matrix. If ``False``, zero will be
-        subtracted.
-    bias : `bool`, optional
-        Whether to use a biased estimate of the number of samples. If ``False``,
-        subtracts ``1`` from the number of samples.
-    inplace : `bool`, optional
+    X : (n_samples, n_dimensions) ndarray
```
Comment (jabooth) [inline menpo/math/decomposition.py]: These docs just need to be formatted with the new doc style

File: menpo/model/pca.py [hunk 2]
```diff
@@ -579,6 +496,56 @@ def orthonormalize_against_inplace(self, linear_model):
...
 
+    def increment(self, samples, n_samples=None, forgetting_factor=1.0,
+                  verbose=False):
+        r"""
+        Update the eigenvectors, eigenvalues and mean vector of this model
+        by performing incremental PCA on the given samples.
+
+        Parameters
+        -----------
+        samples : list of :map:`Vectorizable`
```
Comment (jabooth) [inline menpo/model/pca.py]: and docs here too

File: menpo/model/pca.py [hunk 3]
```diff
@@ -977,3 +944,37 @@ def __str__(self):
             self.noise_variance(), self.noise_variance_ratio(),
             self.n_components, self.components.shape)
         return str_out
+
+
+def extract_data(samples, n_samples=None, verbose=False):
```
Comment (jabooth) [inline menpo/model/pca.py]: @jalabort this is nice how you cleaned it up but in light of @patricksnape s PR #528 we should probably use his generalised solution. Not sure which order we should bring these in?

**Comment by jabooth** [comment]
I'll bring #528 in, merge it with this, and fix the docs.

**Comment by jabooth** [comment]
@jalabort I've merged this with #528 which only changes `extract_data`. I have two tests failing on OS X so we'll see if it's a BLAS thing or a regression I've introduced somehow in the morning. Just the docs to fix then we can get this in.

**Comment by jalabort** [comment]
Hey I had already left when you comment on this yesterday. I guess this is good now since it's been merged...

**Pull request merged**